tests/network/samba/samba_adcli.pm

```diff
@@ -185,6 +192,13 @@ sub post_fail_hook {
     if ($domain_joined) {
         script_run("echo \"\$AD_DOMAIN_PASSWORD\" | net ads leave --domain '$AD_hostname' -U Administrator -i");
     }
+    # Restore the network config files
+    assert_script_run("cp /etc/sysconfig/network/config.bak /etc/sysconfig/network/config");
+    assert_script_run("cp /etc/resolv.conf.bak /etc/resolv.conf");
+}
+
+sub test_flags {
+    return {always_rollback => 1};
```

Comment:
Configuration will rollback when passed or failed.
IMO always_rollback is just redundant, when the test fails due to some backend issue it will not continue at all.